pybop/problems/design_problem.py

```diff
@@ -47,10 +47,13 @@ def __init__(
             signal = ["Voltage [V]"]
         additional_variables.extend(["Time [s]", "Current [A]"])
         additional_variables = list(set(additional_variables))
-        self.warning_patterns = [
-            "Ah is greater than",
-            "Non-physical point encountered",
-        ]
+
+        if init_soc is None:
+            if "Initial SoC" in model._parameter_set.keys():
```

Comment:
It would be easier for me to do this in #424 - where some of the logic regarding modification of the parameter sets is fixed. We can merge this PR into #424 and then #424 into `develop` if you prefer.